core/src/main/resources/hudson/model/Run/console.jelly

```diff
@@ -28,15 +28,32 @@ THE SOFTWARE.
   Displays the console output
 -->
 <?jelly escape-by-default='true'?>
-<st:compress xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
+<st:compress xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:dd="/lib/layout/dropdowns">
   <l:layout title="${it.fullDisplayName} Console">
     <st:include page="sidepanel.jelly" />
-     <l:breadcrumb title="${%Console Output}" />
+
+    <l:breadcrumb title="${%Console Output}" />
+
     <l:main-panel>
-      <t:buildCaption>
+      <j:set var="controls">
+        <l:copyButton ref="out" label="${%Copy}" />
+        <l:overflowButton>
+          <dd:custom>
+            <a class="jenkins-dropdown__item" href="consoleText" download="${it.displayName}.txt">
```

Comment:
Is it really useful to use the displayname here? If I choose a displayname that contains a `/` this wouldn't be saveable.
something
`${it.parent.name}-{it.number}.txt` would guarantee a valid filename

Reply:
> If I choose a displayname that contains a `/` this wouldn't be saveable.

Per https://html.spec.whatwg.org/multipage/links.html#attr-hyperlink-download

> The attribute may have a value; the value, if any, specifies the default filename that the author recommends for use in labeling the resource in a local file system. There are no restrictions on allowed values, but authors are cautioned that most file systems have limitations with regard to what punctuation is supported in filenames, and user agents are likely to adjust filenames accordingly.

Trying this PR and DOM editing in Firefox, I get the file `foo_bar_#1.txt` from the attribute value `foo/bar/#1.txt`.

(`parent.fullName` might still be better though. Unsure whether we should go for display names, there's arguments for and against IMO)